tests/opentelemetry-docker-tests/tests/sqlalchemy_tests/mixins.py

```diff
@@ -199,3 +201,45 @@ def test_parent(self):
         self.assertEqual(parent_span.instrumentation_info.name, "sqlalch_svc")
 
         self.assertEqual(child_span.name, "SELECT " + self.SQL_DB)
+
+    def test_multithreading(self):
+        """Ensure spans are captured correctly in a multithreading scenario
+        
+        We also expect no logged warnings about calling end() on an ended span.
+        """
+
+        def insert_player(session):
+            _session = session()
+            player = Player(name="Player")
+            _session.add(player)
+            _session.commit()
+            _session.query(Player).all()
+
+        def insert_players(session):
+            _session = session()
+            players = []
+            for player_number in range(3):
+                players.append(Player(name=f"Player {player_number}"))
+            _session.add_all(players)
+            _session.commit()
+
+        session_factory = sessionmaker(bind=self.engine)
+        # pylint: disable=invalid-name
+        Session = scoped_session(session_factory)
+        thread_one = threading.Thread(target=insert_player, args=(Session,))
+        thread_two = threading.Thread(target=insert_players, args=(Session,))
+
+        logger = logging.getLogger("opentelemetry.sdk.trace")
+        with self.assertLogs(logger, level="WARNING") as cm:
+            # Dummy warning so test doesn't fail on no logs
+            logger.warning("Dummy warning")
```

Comment:
Can you explain why this assertLogs is needed if you're intentionally creating a warning as part of the test?

Reply:
Yeah, I wanted to be able to assert no warning logs are generated. There would be a warning about using a closed span before the multithreading fix that was merged a while back. If there are no logs when using `assertLogs`, it fails. Is there a preferred alternative when using `unittest`? Mocking the logger instead?

Reply:
I would do this:

``` python
from unittest import TestCase


class Test(TestCase):

    def test_case(self):

        with self.assertRaises(AssertionError):
            with self.assertLogs():
                pass
```